source/sixve.f90

```diff
@@ -211,397 +211,365 @@ subroutine envarsv(dpsv,oidpsv,rvv,ekv)
   use mod_alloc
 
   implicit none
-  integer ih1,ih2,j,kz1,l,l1,l2
-
-  !Local version of variables normally found in mod_common_main
-  real(kind=fPrec) aek,afok,as3,as4,as6,co,dpd,dpsq,dpsv,fi,    &
-        fok,fok1,fokqv,g,gl,hc,hi,hi1,hm,hp,hs,oidpsv,rho,rhoc,rhoi, &
-        rvv,si,siq,sm1,sm12,sm2,sm23,sm3,wf,wfa,wfhi
 
   real(kind=fPrec), allocatable, intent(inout) :: ekv(:,:) !(npart,nele)
 
-  dimension fokqv(npart),dpsv(npart)
-  dimension rvv(npart),oidpsv(npart)
-  dimension dpd(npart),dpsq(npart),fok(npart),rho(npart)
-  dimension fok1(npart),si(npart),co(npart),g(npart),gl(npart)
-  dimension sm1(npart),sm2(npart),sm3(npart),sm12(npart)
-  dimension as3(npart),as4(npart),as6(npart),sm23(npart)
-  dimension rhoc(npart),siq(npart),aek(npart),afok(npart)
-  dimension hp(npart),hm(npart),hc(npart),hs(npart),wf(npart)
-  dimension wfa(npart),wfhi(npart),rhoi(npart)
-  dimension hi(npart),fi(npart),hi1(npart)
+  integer ih1,ih2,j,kz1,l,l1,l2
 
-  real(kind=fPrec) fokm
+  ! Local version of variables normally found in mod_common_main
+  real(kind=fPrec) dpd(npart),dpsq(npart),dpsv(npart),fokqv(npart),oidpsv(npart),rvv(npart)
+  real(kind=fPrec) aek,afok,as3,as4,as6,co,fi,fok,fok1,g,gl,hc,hi,hi1,hm,hp,hs,rho,rhoc,rhoi,&
+    si,siq,sm1,sm12,sm2,sm23,sm3,wf,wfa,wfhi,fokm
 
-!-----------------------------------------------------------------------
   save
```

Comment:
Do we need to `save` this, thus keeping the arrays always-in-memory in case the function gets called again?

Also, isn't `npart` dynamic now? How does that work when you declare a static e.g. `dpd(npart)`?

Reply:
I assumed the save was needed for the remaining arrays, which is only a few now, so it's not much of an issue. Although I could test if removing save blows up.